inspirehep/modules/theme/templates/inspirehep_theme/format/record/Inspire_HTML_detailed_macros.tpl

```diff
@@ -179,9 +179,9 @@
     {% endfor %}
   {% endif %}
 
-  {% if record.get('external_system_numbers') %}
-    {% for system_number in record.get('external_system_numbers') if system_number.get('obsolete') == False and system_number.get('institute', '') | is_institute %}
```

Comment:
@michamos is there an equivalent now of `obsolete` or `institute` that should be skipped when displaying external URLs?

Reply:
not as explicit flags. The logic is now that the first one with a given schema is to be displayed, the others are discarded. So
```python
{
    'external_system_numbers': [
        {
            'schema': 'foo',
            'value': 'val1',
        },
        {
            'schema': 'bar',
            'value': 'val2',
        },
        {
            'schema': 'foo',
            'value': 'val3',
        }
    ]
}
```
should display the first two elements only, because `foo` is repeated.

It's probably good to write a reusable getter that returns the filtered list to be displayed, and while you are at it refactor this to compute the links in python code instead of in the templates.

Reply:
@Dinika in this file https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/records/wrappers.py#L43 you can create a new property for that class to get the external_system_identifiers.

This property will filter the list keeping the order but removing any duplicated schema.

Then in the template you can iterate over it with `record. external_system_identifiers`

Reply:
This last comment is still missing in order to avoid showing duplicates.